sectionfilter.c

```diff
@@ -212,25 +212,39 @@ void cSatipSectionFilter::Process(const uint8_t* dataP)
      }
 }
 
-bool cSatipSectionFilter::Send(void)
+bool cSatipSectionFilter::TransferData(int fd, const uchar* data, int count, int timeoutMs)
+{
+  struct pollfd p;
+  p.fd = fd;
+  p.events = POLLOUT;
+  p.revents = 0;
+
+  while (true) {
+    if (poll(&p, 1, timeoutMs) <= 0)
+       return false;
+    if(send(fd, data, count, MSG_EOR) >= 0)
+      break;
+    if (errno != EAGAIN)
+      return false;
+    }
+
+  return true;
+}
+
+void cSatipSectionFilter::Send(void)
 {
-  bool result = false;
   cFrame *section = ringBufferM->Get();
   if (section) {
      uchar *data = section->Data();
      int count = section->Count();
      if (data && (count > 0) && (socketM[1] >= 0) && (socketM[0] >= 0)) {
-        ssize_t len = send(socketM[1], data, count, MSG_EOR);
-        ERROR_IF(len < 0 && errno != EAGAIN, "send()");
-        if (len > 0) {
-           ringBufferM->Drop(section);
-           result = !!ringBufferM->Available();
+        if (TransferData(socketM[1], data, count, 2000)) {
```

Comment:
The timeout is way too big as it blocks data transfers to all other filters as well. Any polling logic should be implemented into the cSatipSectionFilterHandler and keep all filter file handles in it at the same time.